internal/tmpdir/tmpdir.go

```diff
@@ -15,10 +15,15 @@ var unixTempDirForBigFiles = builtinUnixTempDirForBigFiles
 // DO NOT change this, instead see unixTempDirForBigFiles above.
 const builtinUnixTempDirForBigFiles = "/var/tmp"
 
-// TemporaryDirectoryForBigFiles returns a directory for temporary (big) files.
+// SetTemporaryDirectoryForBigFiles overwrites default temporary directory.
+func SetTemporaryDirectoryForBigFiles(customTmpDir string) {
+	unixTempDirForBigFiles = customTmpDir
```

Comment:
This shouldn’t be a per-library global state, _especially_ not set on per-call private state like `types.SystemContext`.

Instead, just add a `*types.SystemContext` parameter to the `TemporaryDirectoryForBigFiles` function. (That is safe to do because the `internal` package is not a part of the API contract.)